fog/distribution/src/main.rs

```diff
@@ -206,13 +208,11 @@ fn main() {
                 );
 
                 // Push to queue
-                spendable_txouts_sender
-                    .send(SpendableTxOut {
-                        tx_out: tx_out.clone(),
-                        amount: input_amount,
-                        from_account_key: account.clone(),
-                    })
-                    .expect("failed sending to spendable_txouts_sender");
+                spendable_txouts.push(SpendableTxOut {
```

Comment:
This is the main idea- instead of sending the SpendableTxOuts using the sender / receiver structs for the slam threads, create a vector of SpendableTxOuts to be used in two distinct phases (1) Pre-slam debit and (2) slam phase.